cms/src/test/java/uk/nhs/digital/apispecs/ApiSpecificationImportMetadataTest.java

```diff
@@ -0,0 +1,115 @@
+package uk.nhs.digital.apispecs;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static uk.nhs.digital.apispecs.ApiSpecificationImportMetadata.Item.item;
+import static uk.nhs.digital.apispecs.ApiSpecificationImportMetadata.metadataWith;
+
+import org.junit.Test;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class ApiSpecificationImportMetadataTest {
+
+    public static final Comparator<ApiSpecificationImportMetadata.Item> BY_JCR_ID = Comparator.comparing(ApiSpecificationImportMetadata.Item::apiSpecJcrId);
+
+    @Test
+    public void with_createsNewInstancePopulatedWithGivenArguments() {
+
+        // given
+        final List<ApiSpecificationImportMetadata.Item> expectedItems = asList(
+            item("6c684d98-61e9-4ff6-bf40-5e288bd692f8", Instant.parse("2021-11-26T12:30:01Z")),
+            item("06bf9af3-e63d-42cf-a60a-b80063129638", Instant.parse("2021-11-26T12:30:02Z")),
+            item("b214ce0f-8202-46b8-895b-24129a2dfd9c", Instant.parse("2021-11-26T12:30:03Z"))
+        );
+
+        // when
+        final ApiSpecificationImportMetadata actualApiSpecificationImportMetadata = metadataWith(expectedItems);
+
+        // then
+        final List<ApiSpecificationImportMetadata.Item> actualItems = new ArrayList<>(actualApiSpecificationImportMetadata.items());
+
+        // we don't care about the ordering
+        expectedItems.sort(BY_JCR_ID);
+        actualItems.sort(BY_JCR_ID);
+
+        assertThat(
+            "Metadata contains all the items used for its creation.",
+            actualItems,
+            is(expectedItems)
+        );
+    }
+
+    @Test
+    public void with_createsEmptyInstanceWhenNoMetadataItemsGiven() {
+
+        // given
+        final List<ApiSpecificationImportMetadata.Item> expectedItems = emptyList();
+
+        // when
+        final ApiSpecificationImportMetadata actualApiSpecificationImportMetadata = metadataWith(expectedItems);
+
+        // then
+        final List<ApiSpecificationImportMetadata.Item> actualItems = new ArrayList<>(actualApiSpecificationImportMetadata.items());
+
+        assertThat(
+            "Metadata created as empty returns empty collection of items.",
+            actualItems,
+            is(emptyList())
```

Comment:
you're not actually using `expectedItems` in your assertion which i assume was your intent

Reply:
Hmm, indeed, I'm not making _direct_ use of them, which makes the naming a bit confusing.

Admittedly, the intent was to expect an empty list. I'll clean it up.